tests/lib/managers/folders-test.js

```diff
@@ -276,4 +276,19 @@ describe('Folders', function() {
 		});
 	});
 
+	describe('restoreFolder()', function() {
+
+		it('should make POST call to restore a folder', function() {
+			sandbox.stub(boxClientFake, 'defaultResponseHandler');
+			sandbox.mock(boxClientFake).expects('post').withArgs('/folders/1234');
+			folders.restoreFolder(FOLDER_ID, 'Folder Restored', 0);
+		});
+
+		it('should call BoxClient defaultResponseHandler method with the callback when response is returned', function(done) {
+			sandbox.mock(boxClientFake).expects('defaultResponseHandler').withArgs(done).returns(done);
+			sandbox.stub(boxClientFake, 'post').yieldsAsync();
+			folders.restoreFolder(FOLDER_ID, 'Folder Restored', 0, done);
+		});
+	});
```

Comment:
Yes, we should exercise all the code paths:
- When both optional parameters are passed
- When just name is passed
- When just parentfolderID is passed
- When neither optional parameter is passed